src/Tester/DependencyResolverGrainTests.cs

```diff
@@ -0,0 +1,130 @@
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System.IO;
+using System.Threading.Tasks;
+using Autofac;
+using Autofac.Builder;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Orleans.Autofac;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.MembershipService;
+using Orleans.Runtime.ReminderService;
+using Orleans.Streams;
+using Orleans.TestingHost;
+using UnitTests.GrainInterfaces;
+using UnitTests.Grains;
+using UnitTests.Tester;
+
+namespace UnitTests.General
+{
+    /// <summary>
+    /// Summary description for SimpleGrain
+    /// </summary>
+    [DeploymentItem("OrleansDependencyResolverConfigurationForTesting.xml")]
+    [DeploymentItem("Orleans.Autofac.dll")]
+    [TestClass]
+    public class DependencyResolverGrainTests : UnitTestSiloHost
+    {
+        private const string SimpleGrainNamePrefix = "UnitTests.Grains.SimpleG";
+
+        public DependencyResolverGrainTests()
+            : base(new TestingSiloOptions
+            {
+                StartPrimary = true,
+                StartSecondary = false,
+                SiloConfigFile = new FileInfo("OrleansDependencyResolverConfigurationForTesting.xml")
+            })
+        {
+        }
+
+        public ISimpleGrain GetSimpleGrain()
+        {
+            return GrainFactory.GetGrain<ISimpleGrain>(GetRandomGrainId(), SimpleGrainNamePrefix);
+        }
+
+        private static int GetRandomGrainId()
+        {
+            return random.Next();
+        }
+
+        [ClassCleanup]
+        public static void MyClassCleanup()
+        {
+            StopAllSilos();
+        }
+
+        [TestMethod, TestCategory("BVT"), TestCategory("Functional")]
+        public async Task DISimpleGrainGetGrain()
+        {
+            ISimpleGrain grain = GetSimpleGrain();
+            int ignored = await grain.GetAxB();
+        }
+    }
+
+    public class AutofacDepdendencyResolverProvider : IDependencyResolverProvider
+    {
+        public string Name
+        {
+            get { return "Autofac DI"; }
+        }
+
+        public Task Init(string name, IProviderRuntime providerRuntime, IProviderConfiguration config)
+        {
+            //
+            // This method will not be called by the DependenceResolverProviderManager, when loading the provider.
+            //
+
+            throw new System.NotSupportedException();
+        }
+
+        public IDependencyResolver GetDependencyResolver(ClusterConfiguration config, NodeConfiguration nodeConfig, TraceLogger logger)
+        {
+            var builder = new ContainerBuilder();
+
+            builder.RegisterType(typeof(GrainBasedMembershipTable));
```

Comment:
Since this class need to be implemented in every project where you're using the DI container of your choice, type registration can not be omitted, but we can have a base class for Autofac Provider which takes care of the registration of the system grains in an overridable manner.

There are possibilities to further abstract and simplify this provider once we later add an Owin like silo startup mechanism.

For now I would not change this, but stating in the documentation that system grain registration is also needed.